api/client/set.go

```diff
@@ -0,0 +1,100 @@
+package client
+
+import (
+	"fmt"
+
+	Cli "github.com/docker/docker/cli"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/units"
+	"github.com/docker/docker/runconfig"
+)
+
+// CmdSet sets resources of one or more containers.
+//
+// Usage: docker set [OPTIONS] CONTAINER [CONTAINER...]
+func (cli *DockerCli) CmdSet(args ...string) error {
+	cmd := Cli.Subcmd("set", []string{"CONTAINER [CONTAINER...]"}, Cli.DockerCommands["set"].Description, true)
+	flBlkioWeight := cmd.Uint16([]string{"-blkio-weight"}, 0, "Block IO (relative weight), between 10 and 1000")
+	flCPUPeriod := cmd.Int64([]string{"-cpu-period"}, 0, "Limit CPU CFS (Completely Fair Scheduler) period")
+	flCPUQuota := cmd.Int64([]string{"-cpu-quota"}, 0, "Limit CPU CFS (Completely Fair Scheduler) quota")
+	flCpusetCpus := cmd.String([]string{"-cpuset-cpus"}, "", "CPUs in which to allow execution (0-3, 0,1)")
+	flCpusetMems := cmd.String([]string{"-cpuset-mems"}, "", "MEMs in which to allow execution (0-3, 0,1)")
+	flCPUShares := cmd.Int64([]string{"#c", "-cpu-shares"}, 0, "CPU shares (relative weight)")
+	flMemoryString := cmd.String([]string{"m", "-memory"}, "", "Memory limit")
+	flMemoryReservation := cmd.String([]string{"-memory-reservation"}, "", "Memory soft limit")
+	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+	flKernelMemory := cmd.String([]string{"-kernel-memory"}, "", "Kernel memory limit")
+	cmd.Require(flag.Min, 1)
+
+	cmd.ParseFlags(args, true)
```

Comment:
I do: `docker set 5c67` it will send a request to the server but nothing will happen, so this turns into a test to see if a container with that ID exists or not. I'm thinking we should generate an error if no flags are provided - we should be able to check this via: `cmd.NFlag() == 0` I think.  What do people think?